### PR TITLE
Expand docs

### PR DESCRIPTION
**Pull request by yardasol**
## Summary of changes

This PR updates/adds new pages to the online documentation to help users understand how to use the software:
- A page on input and out file format/structure
- Change references to serpent to more general "depletion codes" where applicable
- Updates the installation page with new software dependencies.
- Explicitly adds the `openmc` package to `doc-environment.yml`
- Various typo fixes related to errors when building documentation.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Required for Merging
- [x] #132  has been merged
  - [x] Any merge conflicts that appear have been resolved
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] My change is a source code change
  - [ ] I have added/modified tests to cover my changes
  - [ ] I have explained why my change does not require new/modified tests 
- [x] All new and existing tests passed.
  - [x] CI tests pass
  - [x] Local tests pass (including Serpent2 integration tests)
- [ ] ~I have recorded my changes in the changelog for the upcoming release~ no changes to record

## Associated Issues and PRs

- Issue: Partially addresses #151 

## Associated Developers

- Dev: @

## Checklist for Reviewers

Reviewers should use [this link](https://arfc.github.io/manual/guides/pull_requests) to get to the 
Review Checklist before they begin their review.

### CODE REVIEW

File: doc/installation.rst
Comment (gwenchee) [inline doc/installation.rst:80]:
```suggestion
SaltProc can use any cross section library that its depetion codes can. Currently supported depletion codes and their guides on cross section confguration are listed below:
```

**Review by gwenchee — APPROVED** [review]
LGTM! I left some minor comments.

File: doc/devguide/contributing.rst
Comment (munkm) [inline doc/devguide/contributing.rst:38]: Is this still true? Because Python 3.5 is no longer supported we may want to bump this.

File: doc/devguide/contributing.rst
Comment (munkm) [inline doc/devguide/contributing.rst:91]: Because the official pytest docs use `pytest` as the recommended command, we may want to revise this to be in line with them. 
```suggestion
-  **Make sure the test suite passes** on your computer. To do so, run ``pytest saltproc`` in the base directory of this repository. At a minimum, you must run the tests requiring serpent locally as they are not tested by our CI
```

File: doc/doc-environment.yml
Comment (munkm) [inline doc/doc-environment.yml:7]: Just to confirm: we are not limited on openmc versions as a requirement? (I know this is the doc environment)

File: doc/overview.rst
Comment (munkm) [inline doc/overview.rst:10]: Does this point mean just fuel geometry? Or problem geometry?

File: doc/overview.rst
Comment (munkm) [inline doc/overview.rst:21]: When I think of running a depletion code, I wouldn't consider the Monte Carlo transport as part of the depletion step. Maybe this could be clarified here by "runs the Monte Carlo transport and depletion" or something?

File: doc/overview.rst
Comment (munkm) [inline doc/overview.rst:24]:
```suggestion
  - creates a new input file for Serpent or OpenMC
```

File: doc/overview.rst
Comment (munkm) [inline doc/overview.rst:77]: I love this! Maybe we could link to our issues that are tagged as features or wishlist items here too? What do you think?

File: saltproc/openmc_deplete.py
Comment (munkm) [inline saltproc/openmc_deplete.py:72]: isn't the double asterisk a convention for kwargs?

**Review by munkm — COMMENTED** [review]
Hey @yardasol! Great additions to the docs. Thank you so much for these improvements and clarifications. I have a few questions in my review, but am happy to merge once they're resolved or answered.

File: doc/devguide/contributing.rst
Comment (yardasol) [reply doc/devguide/contributing.rst:91]: Good catch

File: doc/doc-environment.yml
Comment (yardasol) [reply doc/doc-environment.yml:7]: Correct.

File: doc/overview.rst
Comment (yardasol) [reply doc/overview.rst:21]: That's a good point. I'll add that.

File: doc/overview.rst
Comment (yardasol) [reply doc/overview.rst:24]: better to be specific for now

File: doc/devguide/contributing.rst
Comment (yardasol) [reply doc/devguide/contributing.rst:38]: I agree. Let's bump it to 3.7?

File: doc/overview.rst
Comment (yardasol) [reply doc/overview.rst:24]: Actually, I think leaving this generic is fine right now.

File: doc/index.rst
Comment (gwenchee) [reply doc/index.rst]: i don't think you addressed this?

**Comment by gwenchee** [comment]
@yardasol I think you didn't address some of my comments. I'll merge after you do

File: doc/fileformatspec/databasefile.rst
Comment (yardasol) [reply doc/fileformatspec/databasefile.rst]: Sometimes the GH reviewer is a bit buggy and fails to catch changes. I addressed this in e51c68f

**Comment by yardasol** [comment]
> @yardasol I think you didn't address some of my comments. I'll merge after you do

@gwenchee I think the GitHub review interface is buggy sometimes and doesn't catch all the commits, but I double checked and did indeed address all of your comments. Please check again and wait for Madicken to approve before merging.

**Review by gwenchee — APPROVED** [review]
@yardasol It looks good!

File: doc/overview.rst
Comment (munkm) [reply doc/overview.rst:10]: ok can you edit it to clarify that? The way I'm reading this I interpret it as fuel geometry

File: doc/overview.rst
Comment (munkm) [inline doc/overview.rst:29]:
```suggestion
  1. Runs the transport-depletion code (`saltproc.depcode.run_depcode()`)
```

File: saltproc/app.py
Comment (munkm) [inline saltproc/app.py:345]: Did you mean to remove the starting backtick here?

File: saltproc/app.py
Comment (munkm) [inline saltproc/app.py:357]: and here

File: saltproc/app.py
Comment (munkm) [inline saltproc/app.py:367]: and here

**Review by munkm — COMMENTED** [review]
I think maybe some backtick formatting got removed by accident?

**Comment by yardasol** [comment]
Good catches @munkm. I've addressed your comments.

**Review by munkm — APPROVED** [review]

**Comment by munkm** [comment]
Thank you so much @yardasol ! These are great improvements to the docs.

**Pull request merged**
